sceptre/template.py

```diff
@@ -132,9 +134,47 @@ def upload_to_s3(
 
         return url
 
-    def _create_bucket(
-            self, region, bucket_name, connection_manager
-    ):
+    def _bucket_exists(self, bucket_name, connection_manager):
+        """
+        Checks if the bucket ``bucket_name`` exists.
+
+        This is done in a thread-safe way. No error is raised if the bucket
+        already exists.
+
+        :param bucket_name: The name of the bucket to check.
+        :type bucket_name: str
+        :param connection_manager: The connection manager used to make
+            AWS calls.
+        :type connection_manager: sceptre.connection_manager.ConnectionManager
+        :returns: Boolean whether the bucket exists
+        :rtype: bool
+        :raises: botocore.exception.ClientError
+
+        """
+        try:
```

Comment:
I think it's conventional for Python `try` blocks to only contain the code we're expecting might fail. Could we switch this to:

```Python
log("Attempting to find bucket")
try:
    connection_manager.call(...)
except ClientError:
    ...
log("Found template bucket")
return True
```

(I realise this PR is just copy-pasting but I think we might as well fix this now)